LUN-2334 Added option to disable cache clears during long operations.

virtosubogdan opened the pull request.

virtosubogdan committed:
LUN-2334 Added option to disable and reenable cache clears.

virtosubogdan committed:
LUN-2334 Do not perform delete operations when it is not necessary.

virtosubogdan committed:
LUN-2334 Rework for site data cache clear disabling.

Used global variable in settings module instead of attributes on functions.
Provided decorator for this use case to prevent incorrect use.

virtosubogdan committed:
LUN-2334 Prevent cache clears during site deletion.

earthboundkid commented on cms/admin/__init__.py:
```diff
@@ -2,10 +2,11 @@
 import pageadmin
 import useradmin
 import permissionadmin
+import siteadmin
```
Relative imports should be written as `from . import siteadmin`.

virtosubogdan committed:
LUN-2334 Changed cache deletion to improve performance.

-replace multiple cache.delete with cache.delete_many
-reduce the size of query result used to delete user cache keys

virtosubogdan committed:
LUN-2334 Rewrite relative import.
